### PR TITLE
firt commit to new branch

### PR DESCRIPTION
**Pull request by meazabe**

### CODE REVIEW

File: areaOfShapes.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+
+const triangle = {type: "Triangle", sides: {base: 4, height: 6}};
+            
+const rectangle = {type: "Rectangle", sides: {width: 5, height: 6}};
+
+const circle = {type: "Circle", sides: {pi: 3.1415926535897931, radius: 4}};   
+
+const square = {type: "Square", sides: {side: 6}};
+
```
Comment (mekbib-tekle) [inline areaOfShapes.js:18]: you have to always assume the function does not know about the variables declared outside of it, so avoid using those variables inside the function.

Comment (mekbib-tekle) [inline areaOfShapes.js:18]: the if statements should compare shape.type === "triangle" etc

File: calculator.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Write a simple calculator program that calculates 
+the difference, sum or product of two numbers. 
+
+The function takes as a parameter, the type of calculation to perform and two numbers.
+
+Then based on the type of calculation given 
+(addition, subtraction, multiplication, division), 
+it performs the calculation and returns the result. 
+*/
+//const c = '+';
```
Comment (mekbib-tekle) [inline calculator.js]: it's a good practice to always use descriptive variables whenever possible.
here using a & b is ok, but c could be operation

File: discount.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/*
+Write a function that takes price and discount percentage as parameters and 
+calculates the discounted price. 
+The function should give an error message if the price or percentage are < 0. 
+(e.g. If price = 30€, discount = 50%, new price should be 15€)
+
+15 = 30 - (30 * (50 / 100));
+*/
+
```
Comment (mekbib-tekle) [inline discount.js]: This function is just ok, but for the future you could write it shorter like this..

```
function discountCalculator(price, discount) {
    if (price >= 0 && discount >= 0) {
        return price - (price * (discount/100));
    } 

    return "error";
}
```

File: leapYear.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+How to determine whether a year is a leap year 
+To determine whether a year is a leap year, follow these steps:
+1. If the year is evenly divisible by 4, go to step 2. Otherwise, go to step 5.
+2. If the year is evenly divisible by 100, go to step 3. Otherwise, go to step 4.
+3. If the year is evenly divisible by 400, go to step 4. Otherwise, go to step 5.
+4. The year is a leap year (it has 366 days).
+5. The year is not a leap year (it has 365 days).
+*/
+
+
```
Comment (mekbib-tekle) [inline leapYear.js:12]: I really appreciate how you divide each step in different function. One of the most important principles of writing functions is the single responsibility principle which says one function should be responsible for only one thing. And you're doing it well here. BUT If the function is two lines with variable declaration and return, the could be reduced to just returning the value without declaration, and the function will not have any purpose unless it is being called from many places. If it's not called from many places, you might as well use the value in the place that called the function. e.g.
 result = stepFive(); could be
result = "The year is not a leap year (it has 365 days).";

File: maxMin.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+the functions largest and smallest.
+*/
+const a = 0;
+const b = 0;
+const c = 0;
+function comparator(a,b,c) {
+    function largest(a,b,c) {
+    let result;
+    if (a > b && a > c) {
+        result = a + " is the largest.";
```
Comment (mekbib-tekle) [inline maxMin.js]: result should be a number

File: maxMin.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    } else if (c > a && c > b) {
+        result = c + " is the largest.";
+    }
+    return result;
+    }
+
+function smallest(a,b,c) {
+    let result;
+    if (a < b && a < c) {
+        result = a + " is the smallest and ";
```
Comment (mekbib-tekle) [inline maxMin.js]: same here, result should be a number

File: maxMin.js (hunk 5 above)
Comment (mekbib-tekle) [reply maxMin.js]: you don't have to worry about console logging it, it's enough if it returns a number

File: rna.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+Your function should return the complement of the given strand. 
+
+e.g. if given G, A, T, C the output should be C, T, A, G
+*/
+/*
+Pass an array, loop thru the array, everytime you translate one element, 
+collect it into another array like a basket
+*/
+
+const nucleotides1 = ["C", "G", "A", "T", "G", "T", "C", "A", "G", "A", "T", "C"];
```
Comment (mekbib-tekle) [inline rna.js]: nucleotides1 -> nucleotides

File: shippingCost.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+     } else if (location == 'US' || location == 'Canada') {
+         result = 15;
+     } else {
+         result = 20;
+     }
+     return result;
+ }
+
+ function totalCost(productCost, location) {
+     const shippingCost = shippingCostCalculator(location);
```
Comment (mekbib-tekle) [inline shippingCost.js:26]: Great job separating the shippingCostCalculator

File: leapYear.js (hunk 4 above)
Comment (mekbib-tekle) [reply leapYear.js:12]: The function should return a boolean value though

File: areaOfShapes.js (hunk 1 above)
Comment (meazabe) [reply areaOfShapes.js:18]: Of course!

File: calculator.js (hunk 2 above)
Comment (meazabe) [reply calculator.js]: Ok.

File: rna.js (hunk 7 above)
Comment (meazabe) [reply rna.js]: Yea...I was trying to do it another way, then I just forgot to edit the name.

File: areaOfShapes.js (hunk 1 above)
Comment (meazabe) [reply areaOfShapes.js:18]: You gave me a comment on this last time. I forgot to finish it while editing.

File: leapYear.js (hunk 4 above)
Comment (meazabe) [reply leapYear.js:12]: Ok!

**Pull request merged**